addons/project/models/project.py

```diff
@@ -236,17 +236,21 @@ def _compute_attached_docs_count(self):
             project.doc_count = docs_count.get(project.id, 0)
 
     def _compute_task_count(self):
-        task_data = self.env['project.task']._read_group(
+        task_data = self.env['project.task'].with_context(active_test=False)._read_group(
             [('project_id', 'in', self.ids),
              ('is_closed', '=', False)],
-            ['project_id', 'display_project_id:count'], ['project_id'])
+            ['project_id', 'display_project_id:count', 'active:array_agg(active)'], ['project_id'])
         result_wo_subtask = defaultdict(int)
         result_with_subtasks = defaultdict(int)
+        active_task_wo_subtasks = defaultdict(int)
         for data in task_data:
             result_wo_subtask[data['project_id'][0]] += data['display_project_id']
             result_with_subtasks[data['project_id'][0]] += data['project_id_count']
+            nb_of_subtask = data['project_id_count'] - data['display_project_id']
+            active_task_wo_subtasks[data['project_id'][0]] += data['active'].count(True) - nb_of_subtask
+
         for project in self:
-            project.task_count = result_wo_subtask[project.id]
+            project.task_count = active_task_wo_subtasks[project.id] if project.active else result_wo_subtask[project.id]
```

Comment:
Wouldn't it be easier to perform and read if we do two separated read_group (on for the non archived project and one for the archived project)? I have the impression that `active_task_wo_subtasks[data['project_id'][0]] += data['active'].count(True) - nb_of_subtask` could return odd results.